crmsh/sbd.py

```diff
@@ -3,14 +3,263 @@
+        if not utils.service_is_active("pacemaker.service"):
+            return default
+        value = utils.get_property("stonith-watchdog-timeout")
+        return int(value.strip('s')) if value else default
+
+    def _load_configurations(self):
+        """
+        Load necessary configurations for both disk-based/disk-less sbd
+        """
+        self.two_node_without_qdevice = utils.is_2node_cluster_without_qdevice(self.removing)
+
+        dev_list = SBDManager.get_sbd_device_from_config()
+        if dev_list:  # disk-based
+            self.disk_based = True
+            self.msgwait = SBDTimeout.get_sbd_msgwait(dev_list[0])
+            self.pcmk_delay_max = utils.get_pcmk_delay_max(self.two_node_without_qdevice)
+        else:  # disk-less
+            self.disk_based = False
+            self.sbd_watchdog_timeout = SBDTimeout.get_sbd_watchdog_timeout()
+            self.stonith_watchdog_timeout = SBDTimeout.get_stonith_watchdog_timeout()
+
+        self.sbd_delay_start_value = self.get_sbd_delay_start_value()
+        logger.debug("Inspect SBDTimeout: %s", vars(self))
+
+    def get_stonith_timeout(self):
+        """
+        Get stonith-timeout value for sbd cases, formulas are:
+
+        stonith-timeout = 1.2 * (pcmk_delay_max + msgwait) # for disk-based sbd
+        stonith-timeout = 1.2 * max (stonith_watchdog_timeout, 2*SBD_WATCHDOG_TIMEOUT) # for disk-less sbd
+        """
+        if self.disk_based:
+            value = int(1.2*(self.pcmk_delay_max + self.msgwait))
+        else:
+            value = int(1.2*max(self.stonith_watchdog_timeout, 2*self.sbd_watchdog_timeout))
+        logger.debug("Result of SBDTimeout.get_stonith_timeout %d", value)
+        return value
+
+    @classmethod
+    def get_stonith_timeout(cls):
+        cls_inst = cls()
+        cls_inst._load_configurations()
+        return cls_inst.get_stonith_timeout()
+
+    def get_sbd_delay_start_value(self):
+        """
+        Get the value for SBD_DELAY_START, formulas are:
+
+        SBD_DELAY_START = (token + consensus + pcmk_delay_max + msgwait)  # for disk-based sbd
+        SBD_DELAY_START = (token + consensus + 2*SBD_WATCHDOG_TIMEOUT) # for disk-less sbd
+        """
+        value_from_calculation = 0
+        token_consensus_margin = corosync.token_consensus_margin()
+        if self.disk_based:
+            value_from_calculation = token_consensus_margin + self.pcmk_delay_max + self.msgwait
+        else:
+            value_from_calculation = token_consensus_margin + 2*self.sbd_watchdog_timeout
+
+        value_from_configuration = SBDManager.get_sbd_value_from_config("SBD_DELAY_START")
+        if re.search(r'\d+', value_from_configuration):
+            adjust_value = max(int(value_from_configuration), value_from_calculation)
+        else:
+            adjust_value = value_from_calculation
+        return adjust_value
+
+    @staticmethod
+    def get_suitable_sbd_systemd_timeout():
```

Comment:
This function is only called in start_pacemaker() to prepare pacemaker_start_msg. With that, it makes sense to me 
1. to rename to something like, get_sbd_delay_start_sec_from_sysconfig()
2. no need (1.2* )
3. to convert either 1 or yes to 2*SBD_WATCHDOG_TIMEOUT
4. to take the exception of int() to 0, eg. for "no/NO/false/False/xxx"

Reply:
I can't get your means here
> to convert either 1 or yes to 2*SBD_WATCHDOG_TIMEOUT

Whether this function's return value is sbd delay start, or 2*SBD_WATCHDOG_TIMEOUT?

> to take the exception of int() to 0, eg. for "no/NO/false/False/xxx"

I don't understand what does this mean?

Reply:
Changed as 
```
    def get_sbd_delay_start_sec_from_sysconfig():
        """
        Get suitable systemd start timeout for sbd.service
        """
        value = SBDManager.get_sbd_value_from_config("SBD_DELAY_START")
        if value in ["yes", "1"]:
            return 2*SBDTimeout.get_sbd_watchdog_timeout()
        return int(value)
```
This function will not be called in `start_pacemaker` if SBD_DELAY_START set to "no"

Reply:
For get_sbd_delay_start_sec_from_sysconfig(), please at least add a TODO to parse "5ms, 5us, 5s, 5m, 5h" in the future.